src/plonk_recursion.rs

```diff
@@ -305,14 +305,9 @@ fn verify_assumptions<C: Curve, InnerC: HaloEndomorphismCurve<BaseField=C::Scala
     // Compute Z(zeta) f'(zeta) - Z(g * zeta) g'(zeta), which should vanish on H.
     let mut f_prime = one;
     let mut g_prime = one;
-    let quadratic_nonresidues = C::ScalarField::generate_quadratic_nonresidues(NUM_ROUTED_WIRES - 1);
     for i in 0..NUM_ROUTED_WIRES {
-        let s_id = if i == 0 {
-            zeta
-        } else {
-            let k_i = builder.constant_wire(quadratic_nonresidues[i - 1]);
```

Comment:
(I wrote this old code just based on the following, without thinking much about it --

> This can be done by defining `S_ID_1(X) = X`, `S_ID_2(X) = k_1 X`, `S_ID_3(X) = k_2 X`, where `k_1`, `k_2` are quadratic non-residues ∈ `F`.

-- but in hindsight, to be safe I think it should have checked that each `k_i` after the first two was not in a previous coset. Anyway, I think using random shifts is safe and simpler.)